compiler/rustc_infer/src/infer/nll_relate/mod.rs

```diff
@@ -395,12 +394,10 @@ where
 
     fn relate_opaques(&mut self, a: Ty<'tcx>, b: Ty<'tcx>) -> RelateResult<'tcx, Ty<'tcx>> {
         let (a, b) = if self.a_is_expected() { (a, b) } else { (b, a) };
+        let cause = self.delegate.cause().clone();
```

Comment:
why is this clone needed?

Reply:
Ah that's just noise fromthe attempt to pass down the relation itself.